kong/plugins/moesif/log.lua

```diff
@@ -26,31 +26,47 @@ local timer_wakeup_seconds = 1.5
 local gr_helpers = require "kong.plugins.moesif.governance_helpers"
 entity_rules = {}
 
+-- Generates http payload without compression
+-- @param `parsed_url` contains the host details
+-- @param `body`  Message to be logged
+-- @return `payload` http payload
+local function generate_payload_without_compression(parsed_url, application_id, body)
+  local payload = nil
+  payload = string_format(
+    "%s %s HTTP/1.1\r\nHost: %s\r\nConnection: Keep-Alive\r\nX-Moesif-Application-Id: %s\r\nUser-Agent: %s\r\nContent-Type: application/json\r\nContent-Length: %s\r\n\r\n%s",
+    "POST", parsed_url.path, parsed_url.host, application_id, "kong-plugin-moesif/"..plugin_version, #body, body)
+  return payload
+end
+
 -- Generates http payload .
 -- @param `method` http method to be used to send data
 -- @param `parsed_url` contains the host details
 -- @param `message`  Message to be logged
 -- @return `payload` http payload
-local function generate_post_payload(parsed_url, access_token, message, application_id, debug)
+local function generate_post_payload(conf, parsed_url, access_token, message, application_id, debug)
   local payload = nil
   local body = cjson.encode(message)
-  local ok, compressed_body = pcall(compress["CompressDeflate"], compress, body)
-  if not ok then
-    if debug then 
-      ngx_log(ngx_log_ERR, "[moesif] failed to compress body: ", compressed_body)
+  if not conf.disable_moesif_payload_compression then 
+    local ok, compressed_body = pcall(compress["CompressDeflate"], compress, body)
+    if not ok then
+      if debug then 
+        ngx_log(ngx_log_ERR, "[moesif] failed to compress body: ", compressed_body)
```

Comment:
we should reconsider logging entire body, might be huge and flood logs

Reply:
We want to log an entire body for the debugging purpose, since this log will only get printed when debug is set to true, we're fine. In most cases, for the large volume api, debug won't be set to true and it won't flood logs.

Reply:
could body be null/empty, need to skip those before compressing? can we limit logging body to some ceiling size?

Reply:
Nope, body can't be null/empty since it'd be the event being sent to Moesif.